core/base/scalarFieldSmoother/ScalarFieldSmoother.h

```diff
@@ -81,16 +85,26 @@ int ttk::ScalarFieldSmoother::smooth(const triangulationType *triangulation,
   if(!outputData_)
     return -4;
 #endif
-
+  bool useMPI = false;
+#if TTK_ENABLE_MPI
+  if(ttk::isRunningWithMPI() && rankArray != nullptr && globalIds != nullptr)
+    useMPI = true;
+#endif
   SimplexId vertexNumber = triangulation->getNumberOfVertices();
 
   std::vector<dataType> tmpData(vertexNumber * dimensionNumber_, 0);
 
   dataType *outputData = (dataType *)outputData_;
   dataType *inputData = (dataType *)inputData_;
-
+#if TTK_ENABLE_MPI
+  std::unordered_map<SimplexId, SimplexId> gidToLidMap;
+#endif
   // init the output
   for(SimplexId i = 0; i < vertexNumber; i++) {
+#if TTK_ENABLE_MPI
```

Comment:
if gidToLidMap is indeed needed, thanks for moving this to `preconditionDistributedVertices()` (cf above remark)